pkg/webui/lib/selectors/profile-picture.js

```diff
@@ -0,0 +1,71 @@
+// Copyright © 2020 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export const isValidProfilePictureObject = profilePicture =>
+  Boolean(profilePicture) && Boolean(profilePicture.sizes)
+
+export const getProfilePictureSizes = profilePicture => {
+  if (!isValidProfilePictureObject(profilePicture))
+    throw new Error('Invalid profile picture object')
+  return profilePicture.sizes
+}
+
+export const getLargestAvailableProfilePicture = profilePicture => {
+  const sizes = getProfilePictureSizes(profilePicture)
+  const sizeValues = Object.keys(profilePicture.sizes)
+
+  return sizes[0] || sizes[Math.max(...sizeValues)]
+}
+
+export const getSmallestAvailableProfilePicture = profilePicture =>
+  getClosestProfilePictureBySize(profilePicture, 64)
+
+export const getOriginalSizeProfilePicture = profilePicture => {
+  const sizes = getProfilePictureSizes(profilePicture)
+
+  return sizes[0]
+}
+
+export const getClosestProfilePictureBySize = (profilePicture, size) => {
+  const sizes = getProfilePictureSizes(profilePicture)
+  if (sizes[size]) {
+    return sizes[size]
+  }
+  if (sizes.length === 1) {
+    return Object.values(sizes)[0]
```

Comment:
This line returns a size instead of a profile picture

Reply:
The object structure looks like this:

```json

{ 
  sizes: {
    '0': '/assets/blob/profile_pictures/admin.01EXQ1Y3VA3Q2A4CY80WRWTMGJ/original.jpg',
    '64': '/assets/blob/profile_pictures/admin.01EXQ1Y3VA3Q2A4CY80WRWTMGJ/64.jpg',
    '128': '/assets/blob/profile_pictures/admin.01EXQ1Y3VA3Q2A4CY80WRWTMGJ/128.jpg',
     '256': '/assets/blob/profile_pictures/admin.01EXQ1Y3VA3Q2A4CY80WRWTMGJ/256.jpg',
     '512': '/assets/blob/profile_pictures/admin.01EXQ1Y3VA3Q2A4CY80WRWTMGJ/512.jpg'
   }
}
```

So this would return the PP URL, as expected.